src/api/app/models/package.rb

```diff
@@ -120,6 +116,23 @@ def self.check_access?(dbpkg = self)
     return Project.check_access?(dbpkg.project)
   end
 
+  def self.dirty_backend_packages
+    # avoid scope here, it will turn objects readonly
```

Comment:
But a scope just wraps an SQL select. Why wouldn't it be read-only. Besides your implementation also just reads or am I missing something?